src/csv_writer.py

```diff
@@ -20,7 +22,7 @@ def write_frame(waveform_message: dict, csn: str, mrn: str) -> bool:
 
     :return: True if write was successful.
     """
-    sourceSystem = waveform_message.get("sourceSystem", None)
+    sourceStreamId = waveform_message.get("sourceStreamId", None)
```

Comment:
If this is null, the data is essentially useless and we should make a bigger noise about it. I don't know our action should be – perhaps we need a dead-letter exchange?

Reply:
Maybe best for now to raise an exception and stop the controller? This should never happen anyway.

Reply:
I've fixed this with 5f396d755bae
All required waveform fields are checked controller.py and an error logged is any are missing. 
Require fields = `mappedLocationString`, `observationTime`, `sourceStreamId`, `samplingRate`, `unit`, `numericValues`, `mappedLocationString`